### ISSUE TITLE
NullPointerException when trying to deploy Function code via Maven Plugin into existing infrastructure

### ISSUE THREAD

**Issue by juergenbr** [Microsoft/azure-maven-plugins#2052]
### Plugin name and version
azure-functions-maven-plugin
1.18.0
### Plugin configuration in your `pom.xml`
```
<plugins>
            <plugin>
                <groupId>com.microsoft.azure</groupId>
                <artifactId>azure-functions-maven-plugin</artifactId>
                <version>${azure.functions.maven.plugin.version}</version>
                <configuration>
                    <auth>
                        <type>service_principal</type>
                        <client>${CLIENT_ID}</client>
                        <tenant>${tenantid}</tenant>
                        <key>${CLIENT_SECRET}</key>
                        <environment>azure</environment>
                    </auth>
                    <subscriptionId>${subscriptionid}</subscriptionId>
                    <!-- function app name -->
                    <appName>${functionAppName}</appName>
                    <!-- function app resource group -->
                    <resourceGroup>${rgname}</resourceGroup>
                    <region>westeurope</region>
                    <runtime>
                        <!-- runtime os, could be windows, linux or docker-->
                        <os>linux</os>
                        <javaVersion>${java.version}</javaVersion>
                    </runtime>
                    <appSettings>
                        <property>
                            <name>MAIN_CLASS</name>
                            <value>${start-class}</value>
                        </property>
                        <property>
                            <name>FUNCTIONS_EXTENSION_VERSION</name>
                            <value>~4</value>
                        </property>
                    </appSettings>
                </configuration>
                <executions>
                    <execution>
                        <id>package-functions</id>
                        <goals>
                            <goal>package</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```
### Expected behaviour
I want to set up a deployment pipeline to deploy function code to an already existing app service. **The service, function, function slots and app insights configuration are created via Bicep templates and should not be changed via the Maven deployment.** All that is required is to update the function code itself.
### Actual behaviour
NullPointerException during deployment - it seems that the plugin somehow expects an AppInsights configuration which I do not want to provide because all the required configurations are already contained and deployed via Bicep.
```
Caused by: java.lang.NullPointerException
    at com.microsoft.azure.toolkit.lib.appservice.utils.AppServiceUtils.updateDiagnosticConfigurationForWebAppBase (AppServiceUtils.java:366)
    at com.microsoft.azure.toolkit.lib.appservice.function.FunctionAppDraft.lambda$updateResourceInAzure$7 (FunctionAppDraft.java:196)
    at java.util.Optional.ifPresent (Optional.java:183)
    at com.microsoft.azure.toolkit.lib.appservice.function.FunctionAppDraft.updateResourceInAzure (FunctionAppDraft.java:196)
    at com.microsoft.azure.toolkit.lib.appservice.function.FunctionAppDraft.updateResourceInAzure (FunctionAppDraft.java:44)
    at com.microsoft.azure.toolkit.lib.common.model.AbstractAzResourceModule.lambda$update$2 (AbstractAzResourceModule.java:224)
    at com.microsoft.azure.toolkit.lib.common.model.AbstractAzResource.doModify (AbstractAzResource.java:311)
    at com.microsoft.azure.toolkit.lib.common.model.AbstractAzResourceModule.update (AbstractAzResourceModule.java:224)
    at com.microsoft.azure.toolkit.lib.common.model.AbstractAzResourceModule.update (AbstractAzResourceModule.java:48)
    at com.microsoft.azure.toolkit.lib.common.model.AzResource$Draft.updateIfExist (AzResource.java:191)
    at com.microsoft.azure.toolkit.lib.appservice.task.CreateOrUpdateFunctionAppTask.lambda$getUpdateFunctionAppTask$9 (CreateOrUpdateFunctionAppTask.java:164)
    at com.microsoft.azure.toolkit.lib.appservice.task.CreateOrUpdateFunctionAppTask.lambda$registerSubTask$5 (CreateOrUpdateFunctionAppTask.java:106)
    at com.microsoft.azure.toolkit.lib.appservice.task.CreateOrUpdateFunctionAppTask.doExecute (CreateOrUpdateFunctionAppTask.java:260)
    at com.microsoft.azure.maven.function.DeployMojo.createOrUpdateResource (DeployMojo.java:172)
    at com.microsoft.azure.maven.function.DeployMojo.doExecute (DeployMojo.java:97)
    at com.microsoft.azure.maven.AbstractAzureMojo.execute (AbstractAzureMojo.java:525)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:301)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:211)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:165)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:157)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:121)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:127)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.jvnet.hudson.maven3.launcher.Maven35Launcher.main (Maven35Launcher.java:130)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at jenkins.maven3.agent.Maven35Main.launch (Maven35Main.java:178)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:139)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:70)
    at hudson.remoting.UserRequest.perform (UserRequest.java:211)
    at hudson.remoting.UserRequest.perform (UserRequest.java:54)
    at hudson.remoting.Request$2.run (Request.java:376)
    at hudson.remoting.InterceptingExecutorService.lambda$wrap$0 (InterceptingExecutorService.java:78)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:628)
    at java.lang.Thread.run (Thread.java:829)
```
### Steps to reproduce the problem
Deploy using maven and the following goals
`clean package azure-functions:deploy -X`

**Comment by Flanker32** [comment]
@juergenbr Thanks a lot for your report and sorry for the trouble, I'll investigate and fix this issue soon.

**Comment by Flanker32** [comment]
@juergenbr Thanks for your report, the issue should be resolved in azure-functions-maven-plugin 1.19.0, could you please upgrade to latest version and see whether the issue could be reproduced? Many thanks!

**Comment by juergenbr** [comment]
@Flanker32 I was able to test it, now the deployment is working with above configuration using version 1.19.0.
